Explorer/Assets/DCL/PluginSystem/World/SceneUIPlugin.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Arch.SystemGroups;
+using DCL.Optimization.Pools;
+using DCL.PluginSystem.World.Dependencies;
+using DCL.SDKComponents.SceneUI.Components;
+using DCL.SDKComponents.SceneUI.Utils;
+using ECS.ComponentsPooling.Systems;
+using ECS.LifeCycle;
+using System.Collections.Generic;
+using UnityEngine.UIElements;
+using UITextInstantiationSystem = DCL.SDKComponents.SceneUI.Systems.UIText.UITextInstantiationSystem;
+using UITextReleaseSystem = DCL.SDKComponents.SceneUI.Systems.UIText.UITextReleaseSystem;
+using UITransformInstantiationSystem = DCL.SDKComponents.SceneUI.Systems.UITransform.UITransformInstantiationSystem;
+using UITransformParentingSystem = DCL.SDKComponents.SceneUI.Systems.UITransform.UITransformParentingSystem;
+using UITransformReleaseSystem = DCL.SDKComponents.SceneUI.Systems.UITransform.UITransformReleaseSystem;
+using UITransformSortingSystem = DCL.SDKComponents.SceneUI.Systems.UITransform.UITransformSortingSystem;
+using UITransformUpdateSystem = DCL.SDKComponents.SceneUI.Systems.UITransform.UITransformUpdateSystem;
+
+namespace DCL.PluginSystem.World
+{
+    public class SceneUIPlugin : IDCLWorldPluginWithoutSettings
+    {
+        private readonly UIDocument canvas;
+        private readonly IComponentPoolsRegistry componentPoolsRegistry;
+
+        public SceneUIPlugin(
+            UIDocument canvas,
+            StyleSheet canvasStyleSheet,
+            ECSWorldSingletonSharedDependencies singletonSharedDependencies)
+        {
+            this.canvas = canvas;
+            if (this.canvas.rootVisualElement != null)
+            {
+                this.canvas.rootVisualElement.styleSheets.Add(canvasStyleSheet);
+                this.canvas.rootVisualElement.AddToClassList("sceneUIMainCanvas");
+                this.canvas.rootVisualElement.pickingMode = PickingMode.Ignore;
+            }
+
+            componentPoolsRegistry = singletonSharedDependencies.ComponentPoolsRegistry;
+            componentPoolsRegistry.AddComponentPool<VisualElement>(onRelease: UiElementUtils.ReleaseUIElement);
```

Comment:
We should provide an additional configuration of the pool max size, by default it's `10000` and in my opinion, it is way too much for Visual Elements as their memory footprint is huge.
Consider adding a parameter to `AddComponentPool` and configuring it.

Reply:
Good idea, I have added the `maxSize` param to the `AddComponentPool` method. I have set a max size of 200 for `UITransforms` and 100 for `UITexts`, do you think it's ok?

Done here: https://github.com/decentraland/unity-explorer/pull/204/commits/c61537d2faa218c200e73b72d9d0fd740fa022bf